tensorflow/compiler/xla/service/gpu/cudnn_conv_rewriter.cc

```diff
@@ -248,7 +255,63 @@ std::tuple<bool, Window, ConvolutionDimensionNumbers> MatchBackwardFilter(
     backward_conv_dnums.add_kernel_spatial_dimensions(output_spatial_dims[i]);
   }
 
-  return std::make_tuple(true, backward_conv_window, backward_conv_dnums);
+  HloInstruction* lhs = conv->mutable_operand(0);
+  if (conv->feature_group_count() == 1) {
+    return std::make_tuple(true, backward_conv_window, backward_conv_dnums,
+                           lhs);
+  }
+
+  int64 input_batch_dimension = backward_conv_dnums.input_batch_dimension();
+  int64 input_feature_dimension = backward_conv_dnums.input_feature_dimension();
+
+  int64 input_batch = lhs->shape().dimensions(input_batch_dimension);
+  int64 input_feature = lhs->shape().dimensions(input_feature_dimension);
+
+  // Reshape batch_dim G*N -> [G,N]
+  std::vector<int64> reshape_dims = lhs->shape().dimensions();
+  auto num_groups = conv->feature_group_count();
+  // Ensure that input_batch is exact multiple of conv->feature_group_count()
```

Comment:
Although in general comments are good, in this case I think the line of code below says the same thing.
So please either make the comment explain why we ensure it, or remove the comment.